content/blog/2019-10-28-self/index.md

```diff
@@ -0,0 +1,176 @@
++++
+title = "An Efficient Implementation of Self"
+
+[extra]
+bio = """
+  [Rachit Nigam](https://rachitnigam.com) is a second year PhD student interested in
+  programming languages & computer architecture. In his free time, he
+  [subtweets](https://twitter.com/notypes/status/1170037148290080771) his advisor and [annoys tenured professors](https://twitter.com/natefoster/status/1074401015565291520).
+"""
+[[extra.authors]]
+name = "Rachit Nigam"
+link = "https://rachitnigam.com"
++++
+
+[An Efficient Implementation of Self, a Dynamically-Typed Object-Oriented
+Language Based on Prototypes][paper] presents techniques for runtime
+compilation, now more commonly referred to as Just In Time (JIT) compilation,
+for the [Self][] programming language.
+
+> We have developed and implemented techniques that double the performance of
+> the dynamically-typed object oriented languages.
+
+
+## The Challenge
+
+The term "dynamic language" is mostly commonly associated with modern scripting
+languages like Python and JavaScript. Self, a much older language developed
+at the famed Xerox PARC labs takes the philosophy of dynamism to a logical
+extreme -- *everything* in self is a message to an object. This includes
+Java-like method calls on object *as well as* control structures like loops
+and conditionals.
```

Comment:
It might be worth mentioning that JavaScript is directly inspired by Self (and Scheme), which is why there will be some suspicious similarities here.